models.go

```diff
@@ -521,6 +533,12 @@ func (p OfflineQueryParams) WithRequiredOutputs(features ...any) OfflineQueryPar
 	return OfflineQueryParamsComplete{underlying: p.withRequiredOutputs(features...)}
 }
 
+// WithQueryContext returns a copy of Offline Query parameters with the query context added.
+// For use via method chaining. See OfflineQueryParamsComplete for usage examples.
+func (p OfflineQueryParams) WithQueryContext(queryContext *QueryContext) OfflineQueryParamsComplete {
```

Comment:
This shouldn't go from OfflineQueryParams -> OfflineQueryParamsComplete because adding query context doesn't make it "complete", in the sense that it doesn't guarantee that we already have inputs or outputs fields populated. 

Personally, I think for offline query, we could just specify it as a field in `OfflineQueryParams`, and no method is needed.

I feel the same way for OnlineQueryParams actually, because as you found out if we need to do a lot of work if we want to maintain consistency and thread through a `WithQueryContext` or `WithX` for every optional parameter.  For example `CorrelationId` is a field without any such method, and users would just specify `OnlineQueryParams{CorrelationId: "abc"}.WithInputs(...)`  etc. 

Sorry for the back and forth!